src/store/web3wallet.store.ts

```diff
@@ -12,8 +12,24 @@ export const coreAtom = atom(() => {
   });
 });
 
+export const relayerConnectionStatusAtom = atom("disconnected"); // Create a new atom to store the connection status
+
 export const web3walletAtom = atom((get) => {
   const core = get(coreAtom);
+  const [_connectionStatus, setConnectionStatus] = useAtom(
+    relayerConnectionStatusAtom,
+  );
+
+  core.relayer.on("relayer_connect", () => {
+    setConnectionStatus("connected");
+  });
+
+  core.relayer.on("relayer_disconnect", () => {
+    setConnectionStatus("disconnected");
+    core.relayer.restartTransport().catch(() => {
+      console.error("couldn't restart transport")
```

Comment:
We could use a toast maybe to signal the issue to the user, but do you think it makes sense to add the toast when you also have the status ?

Reply:
It would follow the wallet connect design example from: https://docs.walletconnect.com/web3wallet/best_practises#connection-state. 
But then if we show a toast when you go offline, it would make sense to also have it when you go online (otherwise the user might think he's still offline).

Reply:
It's only if the transport cannot restart at all that we want a toast, when we cannot recover

And we should probably try a `restartTransport` when the user clicks on the connection status when offline after a failed restart